src/org/isf/lab/service/LabIoOperationRepository.java

```diff
@@ -5,47 +5,27 @@
-    List<Laboratory> findAllWhereDatesByOrderExamDateDesc(
-            @Param("dateFrom") GregorianCalendar dateFrom,
-            @Param("dateTo") GregorianCalendar dateTo);
-    @Query(value = "SELECT * FROM LABORATORY JOIN EXAM ON LAB_EXA_ID_A = EXA_ID_A "
-    		+ "WHERE LAB_EXAM_DATE >= DATE(:dateFrom) AND LAB_EXAM_DATE <= DATE(:dateTo) AND EXA_DESC = :exam ORDER BY LAB_EXAM_DATE DESC, LAB_ID DESC", nativeQuery= true)
-    List<Laboratory> findAllWhereDatesAndExamByOrderExamDateDesc(
-            @Param("dateFrom") GregorianCalendar dateFrom,
-            @Param("dateTo") GregorianCalendar dateTo,
-            @Param("exam") String exam);
-    
-    @Query(value = "SELECT * FROM (LABORATORY JOIN EXAM ON LAB_EXA_ID_A=EXA_ID_A)" +
-    		 " LEFT JOIN LABORATORYROW ON LABR_LAB_ID = LAB_ID WHERE LAB_PAT_ID = :patient " +
-    		 " ORDER BY LAB_DATE, LAB_ID", nativeQuery= true)
-    List<Laboratory> findAllWherePatientByOrderDateAndId(
-            @Param("patient") Integer patient);
-    
-    
-    @Query(value = "SELECT * FROM (LABORATORY JOIN EXAM ON LAB_EXA_ID_A = EXA_ID_A)" +
-		       " JOIN EXAMTYPE ON EXC_ID_A = EXA_EXC_ID_A" +
-			   " WHERE LAB_EXAM_DATE >= DATE(:dateFrom) AND LAB_EXAM_DATE <= DATE(:dateTo) ORDER BY EXC_DESC", nativeQuery= true)
-    List<Laboratory> findAllWhereDatesForPrint(
-            @Param("dateFrom") GregorianCalendar dateFrom,
-            @Param("dateTo") GregorianCalendar dateTo);
-    @Query(value = "SELECT * FROM (LABORATORY JOIN EXAM ON LAB_EXA_ID_A = EXA_ID_A)" +
-		       " JOIN EXAMTYPE ON EXC_ID_A = EXA_EXC_ID_A" +
-			   " WHERE LAB_EXAM_DATE >= DATE(:dateFrom) AND LAB_EXAM_DATE <= DATE(:dateTo) AND EXA_DESC LIKE %:exam% ORDER BY EXC_DESC", nativeQuery= true)
-    List<Laboratory> findAllWhereDatesAndExamForPrint(
-            @Param("dateFrom") GregorianCalendar dateFrom,
-            @Param("dateTo") GregorianCalendar dateTo,
-            @Param("exam") String exam);
-    
-   // @Query(value="SELECT MAX(LAB_MPROG) FROM LABORATORY WHERE YEAR(LAB_DATE) = :year AND MONTH(LAB_DATE ) = :month ", nativeQuery= true)
-   // public int getProgMonth(@Param("month")int month, @Param("year")int year);
-   
+    List<Laboratory> findByExamDateBetweenOrderByExamDateDesc(
+            GregorianCalendar dateFrom,
+            GregorianCalendar dateTo);
+
+    List<Laboratory> findByExamDateBetweenAndExam_DescriptionOrderByExamDateDesc(
+            GregorianCalendar dateFrom,
+            GregorianCalendar dateTo,
+            String exam);
+
+    List<Laboratory> findByPatient_CodeOrderByRegistrationDate(Integer patient);
+
+    List<Laboratory> findByExamDateBetweenOrderByExam_Examtype_DescriptionDesc(
+            GregorianCalendar dateFrom,
+            GregorianCalendar dateTo);
+
+    List<Laboratory> findByExamDateBetweenAndExam_DescriptionContainingOrderByExam_Examtype_DescriptionDesc(
+            GregorianCalendar dateFrom,
+            GregorianCalendar dateTo,
+            String exam);
+
```

Comment:
same here, are Javadocs required?

Reply:
I don't think so. They would not give any extra value comparing to the method signature.

Reply:
IMHO the standard query naming convention is purposely self-explaining, so no need of Javadoc and also it would be an overhead at this stage.

Reply:
For me is perfect, I'm also getting on board, I was just curious about the rules here, since I have seen different opinions!